Mathlib/LinearAlgebra/Multilinear/TensorProduct.lean

```diff
@@ -22,6 +22,42 @@ variable {R ι₁ ι₂ ι₃ ι₄ : Type*}
 variable [CommSemiring R]
 variable {N₁ : Type*} [AddCommMonoid N₁] [Module R N₁]
 variable {N₂ : Type*} [AddCommMonoid N₂] [Module R N₂]
+
+attribute [local simp] add_tmul tmul_add smul_tmul
+
+section
+
+variable {N : ι₁ ⊕ ι₂ → Type*} [∀ i, AddCommMonoid (N i)] [∀ i, Module R (N i)]
+
+/-- Given a family of modules `N` indexed by a type `ι₁ ⊕ ι₂`,
+a multilinear map from the modules `N (.inl i₁)` to `N₁` and
+a multilinear map from the modules `N (.inr i₁)` to `N₂`, this
+is the induced multilinear map from all the modules `N i` to `N₁ ⊗ N₂`. -/
+@[simps apply]
+def domCoprodDep (a : MultilinearMap R (fun i₁ ↦ N (.inl i₁)) N₁)
```

Comment:
Does `domCoprod_domDomCongr_sumCongr` generalize to this case?

Reply:
It should, but it is unclear what should be the most useful phrasing (because permuting the modules may break definitional properties). Then, I would rather add it in a separate PR only when we need it.